python/polycap.pyx

```diff
@@ -362,16 +451,79 @@ cdef class TransmissionEfficiencies:
     cdef polycap_transmission_efficiencies *trans_eff
     cdef object energies_np
     cdef object efficiencies_np
+    cdef polycap_leak **int_leaks
+    cdef int64_t n_int_leaks
+    cdef polycap_leak **ext_leaks
+    cdef int64_t n_ext_leaks
+    cdef int64_t n_start
+    cdef polycap_vector3 *start_coords
+    cdef polycap_vector3 *start_direction
+    cdef polycap_vector3 *start_elecv
+    cdef polycap_vector3 *src_start_coords
+    cdef int64_t n_exit
+    cdef size_t n_energies
+    cdef polycap_vector3 *exit_coords
+    cdef polycap_vector3 *exit_direction
+    cdef polycap_vector3 *exit_elecv
+    cdef double **exit_weights
+    cdef int64_t *n_refl
+    cdef double *d_travel
 
     def __cinit__(self):
         self.trans_eff = NULL
         self.energies_np = None
         self.efficiencies_np = None
+        self.ext_leaks = NULL
+        self.n_ext_leaks = 0
+        self.int_leaks = NULL
+        self.n_int_leaks = 0
+        self.n_start = 0
+        self.start_coords = NULL
+        self.start_direction = NULL
+        self.start_elecv = NULL
+        self.n_exit = 0
+        self.exit_coords = NULL
+        self.exit_direction = NULL
+        self.exit_elecv = NULL
+        self.src_start_coords = NULL
+        self.exit_weights = NULL
+        self.n_refl = NULL
+        self.d_travel = NULL
 
     def __dealloc__(self):
         '''free a :ref:``TransmissionEfficiencies`` class and all associated data'''
         if self.trans_eff is not NULL:
             polycap_transmission_efficiencies_free(self.trans_eff)
+
+        if self.n_ext_leaks > 0:
+            for i in range(self.n_ext_leaks):
+                polycap_leak_free(self.ext_leaks[i])
+            polycap_free(self.ext_leaks)
+        
+        if self.n_int_leaks > 0:
+            for i in range(self.n_int_leaks):
+                polycap_leak_free(self.int_leaks[i])
+            polycap_free(self.int_leaks)
+            
+        if self.start_coords:
+            polycap_free(self.start_coords)
+        if self.start_direction:
+            polycap_free(self.start_direction)
+        if self.start_elecv:
+            polycap_free(self.start_elecv)
+        if self.src_start_coords:
+            polycap_free(self.src_start_coords)
+
+        if self.n_exit > 0:
```

Comment:
Uw probleem zit in deze if-block. Als je op de lijn ervoor een return zet is er geen segfault. Sorry geen tijd om precies uit te zoeken waar 😢

Reply:
aaah.. Fixed it :D Nu zien of de output goed is :P